game_patch/hud/multi_spectate.cpp

```diff
@@ -204,6 +205,19 @@ bool multi_spectate_execute_action(rf::ControlConfigAction action, bool was_pres
     return false;
 }
 
+void multi_spectate_on_player_death(rf::Player* player, rf::Player* killer)
+{
+    // check follow killer cvar first
+    if (g_spectate_mode_follow_killer) {
+        if (player != rf::local_player && killer != rf::local_player && g_spectate_mode_target == player) {
+            // spectate killer if we were spectating victim
+            // avoid spectating ourselves if we somehow managed to kill the victim
+            multi_spectate_set_target_player(killer);
+            return;
+        }
+    }
+}
```

Comment:
Lol I think you made it request me as reviewer instead of rafalh.

I would rewrite it as follows:

```
void multi_spectate_on_player_kill(rf::Player* killed, rf::Player* killer)
{
    // spectate killer if we were spectating victim
    // avoid spectating ourselves if we somehow managed to kill the victim
    if (g_spectate_mode_follow_killer && 
        g_spectate_mode_target == killed &&
        killer != rf::local_player)  {
            multi_spectate_set_target_player(killer);
        }
    }
}
```

Reply:
If someone leaves a review, my option is to have them re-review :P rafalh hasn't left a review (yet).

On your change - I was going to write it exactly that way but decided against it because I didn't see a need to do additional checks after every single kill in the game if `g_spectate_mode_follow_killer` is false (which it will be in the vast majority of cases). It's more lightweight this way - that was my perspective in writing it, anyway.

Will see what rafalh's take on this is though, I'm good either way, just it made more sense to me to reduce the number of checks being done in the vast majority of cases.

Reply:
It is the same number of checks because boolean logic short-circuits.

> On your change - it's missing the player != rf::local_player, which is needed to have it only happen when you're already in spectate mode.

```
void multi_spectate_on_player_kill(rf::Player* killed, rf::Player* killer)
{
    if (!g_spectate_mode_enabled) {
        return;
    }
    
    // spectate killer if we were spectating victim
    // avoid spectating ourselves if we somehow managed to kill the victim
    if (g_spectate_mode_follow_killer && 
        g_spectate_mode_target == killed &&
        killer != rf::local_player)  {
            multi_spectate_set_target_player(killer);
        }
    }
}
```

Imo it is clearer.

Reply:
> It is the same number of checks because boolean logic short-circuits.

I did not realize that - thanks!

I've committed an optimization.